app/code/Magento/Ui/TemplateEngine/Xhtml/Result.php

```diff
@@ -41,26 +42,34 @@ class Result implements ResultInterface
      * @var LoggerInterface
      */
     protected $logger;
+    
+    /**
+     * @var Json
+     */
+    private $jsonSerializer;
 
     /**
      * @param Template $template
      * @param CompilerInterface $compiler
      * @param UiComponentInterface $component
      * @param Structure $structure
      * @param LoggerInterface $logger
+     * @param Json $jsonSerializer
      */
     public function __construct(
         Template $template,
         CompilerInterface $compiler,
         UiComponentInterface $component,
         Structure $structure,
-        LoggerInterface $logger
+        LoggerInterface $logger,
+        Json $jsonSerializer
```

Comment:
Please, inject new dependencies according to [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.1/contributor-guide/backward-compatible-development/index.html)